docs/src/developer/design/user_api.rst

```diff
@@ -36,133 +55,359 @@ functional-style and cmake-based
+during the "Find Dependencies" step). To that end CMake defines the
+``BUILD_TESTING`` variable; when set to ``TRUE`` tests are built, otherwise they
+are not. Proposed CMaize APIs for testing a project:
+
+.. code-block:: CMake
+
+   cmaize_find_or_build_test_dependency(
+       <name>
+       URL <where_on_the_internet_to_download_from>
+       VERSION <the_version_you_want>
+       BUILD_TARGET <target_to_build>
+       FIND_TARGET <target_representing_package>
+       CMAKE_ARGS <configuration_options_to_set>
+   )
+
+   cmaize_add_test_library(
+       <name>
+       SOURCE_DIR <where_the_source_files_are_located>
+       INCLUDE_DIRS <directories_containing_header_files>
+       DEPENDS <dependency0> <dependency1> ...
+   )
+
+   cmaize_add_test_executable(
+       <name>
+       SOURCE_DIR <where_the_source_files_are_located>
+       INCLUDE_DIRS <directories_containing_header_files>
+       DEPENDS <dependency0> <dependency1> ...
+   )
+
+   # This is actually CTest's add_test command
+   add_test(NAME <name> COMMAND)
+
+   # This is a convenience function for the common scenario where the
+   # add_test call simply calls the executable arising from the
+   # cmaize_add_test_executable
+   cmaize_add_test(
+       <name>
+       SOURCE_DIR <where_the_source_files_are_located>
+       INCLUDE_DIRS <directories_containing_header_files>
+       DEPENDS <dependency0> <dependency1> ...
+   )
+
+All of the test functions are thin wrappers around the non-test functions of
+the same name (e.g., ``cmaize_add_test_library`` wraps ``cmaize_add_library``),
+which hide the logic for including the CTest CMake module, and checking that
+``BUILD_TESTING`` is enabled (if it's not the functions are no-ops).
+
+Install Project
+===============
+
+If the tests are successful (or were skipped) it's on to :term:`package`
+installation. Installation typically requires specifying which targets are
+part of the package, generating the packaging files, and then literally
+moving the targets and files to their final location. The main considerations
+for installing are:
+
+- Collecting sufficient information to be able to install the package including:
+  where it goes, which pieces get installed, and what the runtime dependencies
+  are.
+- Installation should be done in a manner which considers the package manager.
+
+The proposed installation :term:`API` is:
+
+.. code-block:: CMake
+
+   cmaize_add_package(<name> NAMESPACE <namespace>)
+
+Each of the user API calls proceeding ``cmaize_add_package`` record the
+information provided. In turn when it comes time to write the packaging files
+and install the package, CMaize can do so in a largely automatic manner simply
+by inspecting the information which was already provided. The only piece of
+information which has not been been provided yet is the namespace to use in
+the package files (CMake recommends appending a prefix to installed targets to
```

Comment:
Do you have a citation of CMake recommending using namespaces? It would be good to link to something for further reading into this and potentially as a gateway into other recommendations made by CMake.

After a quick search, I found some information [here](https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html) (search for "namespace" on the page), but I'm not sure if it is sufficient. Here is a [link](https://cmake.org/cmake/help/latest/policy/CMP0028.html) to the policy defining double colons (`::`) as a pattern used in namespacing `IMPORTED` and `ALIAS` targets.